numba/targets/linalg.py

```diff
@@ -1261,3 +1266,117 @@ def lstsq_impl(a, b, rcond=-1.0):
         return (x, res, rank, s[:minmn])
 
     return lstsq_impl
+
+# specialisations for np.linalg.solve
+
+
+def _get_solve_compute_return_impl(b):
+    # gets an implementation for extracting 'x' (the solution) from
+    # the 'b' scratch space
+    ndim = b.ndim
+    if ndim == 1:
+        @jit(nopython=True)
+        def oneD_impl(b):
+            return b.T.ravel()
+        return oneD_impl
+    else:
+        @jit(nopython=True)
+        def twoD_impl(b):
+            return b
+        return twoD_impl
+
+
+@overload(np.linalg.solve)
+def solve_impl(a, b):
+    ensure_lapack()
+
+    _check_linalg_matrix(a, "solve")
+    _check_linalg_1_or_2d_matrix(b, "solve")
+
+    a_F_layout = a.layout == 'F'
+    b_F_layout = b.layout == 'F'
+
+    # the typing context is not easily accessible in `@overload` mode
+    # so type unification etc. is done manually below
+    a_np_dt = np_support.as_dtype(a.dtype)
+    b_np_dt = np_support.as_dtype(b.dtype)
+
+    np_shared_dt = np.promote_types(a_np_dt, b_np_dt)
```

Comment:
Does that mean you support e.g. a `float64` A with a `float32` B? I think it would be better to only allow homogenous dtypes, and have the user convert manually if they want/need to.